src/jdk.compiler/share/classes/com/sun/tools/javac/processing/JavacFiler.java

```diff
@@ -275,6 +275,14 @@ public Modifier getAccessLevel() {
             this.fileObject = fileObject;
         }
 
+        @Override
+        public void write(byte b[], int off, int len) throws IOException {
+            if ((off | len | (b.length - (len + off)) | (off + len)) < 0)
+                throw new IndexOutOfBoundsException();
+            out.write(b, off, len);
```

Comment:
If I understand this `out` is the underlying `OutputStream` and calling `write(byte[], int, int)` performs the same one-byte-at-a-time write as without the override.

Reply:
In my opinion, the `out` depends on the `FileObject.openOutputStream` which differ depending on the implementation of `FileObject` and  can be set by users in the implementation of `FileObject`.

Reply:
Ok, got it. So, in case the stream has an efficient write of an array range it currently isn't called, but a 1-byte-at-a-time write inherited is called instead.

I think you can use `Objects::checkFromIndexSize` instead for the index check.